### PR TITLE
Remove our own ProfileKey type and expose the zkgroup one

### PR DESCRIPTION
**Pull request by gferon**
Trying to remove inconsistencies with types from `libsignal` and our own. We only used this one for the `.derive_access_key` method, which I moved into a `ProfileKeyExt` trait instead.

### CODE REVIEW

File: libsignal-service/src/push_service.rs [hunk 1]
```diff
@@ -170,9 +167,13 @@ impl fmt::Debug for HttpAuth {
...
-impl ProfileKey {
-    pub fn derive_access_key(&self) -> Vec<u8> {
-        let key = GenericArray::from_slice(&self.0);
+pub trait ProfileKeyExt {
+    fn derive_access_key(&self) -> Vec<u8>;
+}
+
+impl ProfileKeyExt for ProfileKey {
+    fn derive_access_key(&self) -> Vec<u8> {
+        let key = GenericArray::from_slice(&self.bytes);
```
Comment (rubdos) [inline libsignal-service/src/push_service.rs:176]: Should we make a PR to zkgroup for this?

File: libsignal-service/src/websocket.rs [hunk 2]
```diff
@@ -104,11 +104,12 @@ impl<WS: WebSocketService> SignalWebSocketProcess<WS> {
...
-                self.request_sink.send((request, sink)).await.map_err(
-                    |_| ServiceError::WsError {
+                self.request_sink.send((request, sink)).await.map_err(|e| {
+                    log::error!("websocket request failed: {e}");
+                    ServiceError::WsError {
                         reason: "request handler failed".into(),
-                    },
-                )?;
+                    }
+                })?;
```
Comment (rubdos) [inline libsignal-service/src/websocket.rs:112]: This seems irrelevant for this PR.

File: libsignal-service/src/websocket.rs [hunk 3]
```diff
@@ -291,10 +292,11 @@ impl SignalWebSocket {
             ws,
             stream,
         };
+
         let process = process.run().map(|x| match x {
             Ok(()) => (),
-            Err(e) => {
-                log::error!("SignalWebSocket: {}", e);
+            Err(error) => {
+                log::error!("{error}");
```
Comment (rubdos) [inline libsignal-service/src/websocket.rs:299]: Same, irrelevant to this PR.

**Review by rubdos — CHANGES_REQUESTED** [review]

File: libsignal-service/src/push_service.rs (hunk 1 above)
Comment (gferon) [reply libsignal-service/src/push_service.rs:176]: I think so, but I always try to avoid it 😅

**Comment by gferon** [comment]
@rubdos sorry about the unrelated changes that were me trying to find a heisenbug, this is good to go now!

**Comment by rubdos** [comment]
> EDIT: whisperfish doesn't seem to use these types to store the data, so I'm 99.99% sure things won't break.

Don't worry about breaking Whisperfish. I care about getting this library is good shape :)

File: libsignal-service/src/push_service.rs (hunk 1 above)
Comment (rubdos) [reply libsignal-service/src/push_service.rs:176]: I'll give it a try.

**Review by rubdos — APPROVED** [review]

**Comment by rubdos** [comment]
By the way, no need to wait for signalapp/libsignal#508, that's going to take some time anyway!
